src/ketu/async/source.clj

```diff
@@ -125,7 +126,10 @@
 
             (loop []
               (when-let [records (poll!)]
-                (run! put! records)
+                (if (seq records)
```

Comment:
this means an additional `listTopics` network call every time an empty poll is received. Might have performance/load implications? I doubt it, since it only happens when 0 lag and the call should be pretty lightweight.